pygmt/figure.py

```diff
@@ -124,12 +125,13 @@ def region(self):
         return wesn
 
     @fmt_docstring
+    @deprecate_parameter("icc_gray", "resize", "v0.6.0", remove_version="v0.8.0")
```

Comment:
After the changes in this PR, scripts that use `icc_gray` will see a warning:

```
The 'icc_gray' parameter has been deprecated since v0.6.0 and will be removed in v0.8.0.
Please use 'resize' instead.
```
This warning doesn't make sense, because the 'resize' parameter has nothing to do with ICC profiles, so changing `icc_gray` to `resize` still breaks users old scripts.

I think adding the `resize` parameter should be considered an enhancement. As for `icc_gray`, I don't have a good way to deprecate this parameter without breaking backward compatibility, but perhaps we can change the `deprecate_parameter` decorator to allow a `None` new parameter name.
```
@deprecate_parameter("icc_gray", None, "v0.6.0", remove_version="v0.8.0")
```
This new style should give a warning like:

```
The 'icc_gray' parameter has been deprecated since v0.6.0 and will be removed in v0.8.0.
```

Reply:
Agree with you @seisman. However, this would mean that we need to support both parameters, ``icc_gray`` as well as ``resize`` for a while, right?

Reply:
Perhaps we can change the function definition to:
```
psconvert(self, icc_gray=False, **kwargs):
```
and check if `icc_gray` is set to `True`:
```
if icc_gray:
    warnings.warn("depreaction warning here", category=FutureWarning, stacklevel=2)
    if "N" not in kwargs:
        kwargs["N"] = "+i"
    else: 
        kwargs["N"] += "+i"
```

Reply:
Ok, we also need to add an alias for ``-N``. Currently I named it ``bb_style`` (boundingbox style). Feel free to provide other suggestions.